File: code_tuts/8_Recursion&Dynamic_Programming/javascript_solutions/simple_recursion.js
/*
RECURSIVE FUNCTIONS
Author: Sebastian Opiyo

- A simple recursive function example. 
*/

const fib = (n) => {
    if (n <= 2) return 1;
    return fib(n - 1) + fib(n - 2);
};

// TESTS
console.log(fib(3));
console.log(fib(8));
console.log(fib(16));
console.log(fib(40));
console.log(fib(50));

File: code_tuts/8_Recursion&Dynamic_Programming/python_solutions/simple_recursion.py
'''
RECURSIVE FUNCTIONS
Author: Sebastian Opiyo

- A simple recursive function example. 
'''

def recursive_func(n)-> int:
    if n <= 2:
        return 1
    return recursive_func(n -1 ) + recursive_func(n - 2)


if __name__ == '__main__':
    print(recursive_func(3))
    print(recursive_func(8))
    print(recursive_func(16))
    print(recursive_func(40))
    print(recursive_func(50))